src/Compilers/CSharp/Portable/Errors/ErrorCode.cs

```diff
@@ -2150,13 +2150,13 @@ internal enum ErrorCode
         WRN_RefReturnOnlyParameter2 = 9095,
         ERR_RefAssignValEscapeWider = 9096,
         WRN_RefAssignValEscapeWider = 9097,
-
         #endregion
 
         #region diagnostics introduced for C# 12.0
         ERR_ImplicitlyTypedDefaultParameter = 9098,
         WRN_OptionalParamValueMismatch = 9099,
         WRN_ParamsArrayInLambdaOnly = 9100,
+        ERR_NonConstantConversionInConstantPattern = 9101,
```

Comment:
If we keep the approach of checking for user-defined conversions, then let's change this to `ERR_UserDefinedConversionInConstantPattern`.